plugins/go-send-custom-error-conditional/CustomGoPlugin.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/TykTechnologies/tyk/header"
+	"github.com/TykTechnologies/tyk/log"
+)
+
+var logger = log.Get()
+
+// Your custom function to return the custom error message
+func ReturnError(rw http.ResponseWriter, r *http.Request) {
+
+	//your-custom-code-here
+
+	//if header value != 'foo' return function below
+	//if query param != 'x' return function below
+	//if request is going to a specific target, return the error
+	//based on our redirection rules, if request is redirected to x/y/z upstream, return the custom msg below
+
+	errorHeader := r.Header.Get("errorHeader")
+	logger.Info(fmt.Sprintf("ErrorHeader value is: %s", errorHeader))
+
+	sendErrorResponseFromMiddleware(rw, errorHeader)
+}
+
+// Custom function that contains a list of custom error messages. 
+func sendErrorResponseFromMiddleware(rw http.ResponseWriter, key string) {
+	content := map[string]interface{}{
```

Comment:
I would add a small comment that states that this can be done through the config data per API def